src/yunohost/data_migrations/0019_extends_permissions_features_1.py

```diff
@@ -0,0 +1,163 @@
+
+        logger.info(m18n.n("migration_0019_add_new_attributes_in_ldap"))
+        ldap = _get_ldap_interface()
+        permission_list = user_permission_list(short=True, full_path=False)["permissions"]
+
+        labels = {}
+        for app in _installed_apps():
+            labels[app] = app_setting(app, 'label')
+            app_setting(app, 'label', delete=True)
+
+        for permission in permission_list:
+            if permission.split('.')[0] == 'mail':
+                ldap.update('cn=%s,ou=permission' % permission, {
+                    'authHeader': ["FALSE"],
+                    'label': ['E-mail'],
+                    'showTile': ["FALSE"],
+                    'isProtected': ["TRUE"],
+                })
+            elif permission.split('.')[0] == 'xmpp':
+                ldap.update('cn=%s,ou=permission' % permission, {
+                    'authHeader': ["FALSE"],
+                    'label': ['XMPP'],
+                    'showTile': ["FALSE"],
+                    'isProtected': ["TRUE"],
+                })
+            else:
+                label = labels[permission.split('.')[0]].title()
+
+                if permission.endswith(".main"):
+                    ldap.update('cn=%s,ou=permission' % permission, {
+                        'authHeader': ["TRUE"],
+                        'label': [label],
+                        'showTile': ["TRUE"],
+                        'isProtected': ["FALSE"]
+                    })
+                else:
+                    ldap.update('cn=%s,ou=permission' % permission, {
+                        'authHeader': ["TRUE"],
+                        'label': ["%s (%s)" % (label, permission.split('.')[1])],
+                        'showTile': ["FALSE"],
+                        'isProtected': ["TRUE"]
+                    })
+
+    def migrate_skipped_unprotected_protected_uris(self, app=None):
+
+        logger.info(m18n.n("migration_0019_migrate_old_app_settings"))
+        apps = _installed_apps()
+
+        if app:
+            if app not in apps:
+                logger.error("Can't migrate permission for app %s because it ain't installed..." % app)
+                apps = []
+            else:
+                apps = [app]
+
+        def _get_setting(app, name):
+            s = app_setting(app, name)
+            return s.split(',') if s else []
+
+        for app in apps:
+            skipped_urls = [uri for uri in _get_setting(app, 'skipped_uris') if uri != '/']
+            skipped_urls += ['re:' + regex for regex in _get_setting(app, 'skipped_regex')]
+            unprotected_urls = [uri for uri in _get_setting(app, 'unprotected_uris') if uri != '/']
+            unprotected_urls += ['re:' + regex for regex in _get_setting(app, 'unprotected_regex')]
+            protected_urls = [uri for uri in _get_setting(app, 'protected_uris') if uri != '/']
+            protected_urls += ['re:' + regex for regex in _get_setting(app, 'protected_regex')]
```

Comment:
For these I removed an undefined `_sanitized_absolute_url` which was I believe coming from some old copypasta from app_map / app_ssowat ... My understanding is that in that new context it's not needed anymore (we're not generating the ssowat conf but filling some permission properties) ... but not 100% sure ?

Reply:
Well, I think it was not a copy pas. The idea was probably to be sure that all url that we put in the permission a already sanitized. And as some really old instance still have some "dirty" I think that it's good to run this here to correct anything everywhere.

Reply:
Re-investigating this ... they are later automatically sanitized inside the permission_create (or rather permission_url) so no need to pre-sanitize them I believe